File: Taeyeong/재귀/재귀의 귀재.java
import java.io.*;

public class Main {

    static int count = 0;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());
        for(int i = 0; i < N; i++) {
            String str = br.readLine();
            bw.write(isPalindrome(str) + " " + count + "\n");
            count = 0;
        }
        bw.flush();
    }

    static int recursion(String s, int l, int r) {
        count++;
        if(l >= r) return 1;
        else if(s.charAt(l) != s.charAt(r)) return 0;
        else return recursion(s, l+1, r-1);
    }
    static int isPalindrome(String s) {
        return recursion(s, 0, s.length() - 1);
    }
}

File: Jongmin/2st/재귀의_귀재.py
def recursion(s, l, r, cnt=0):
    if l >= r: return 1, cnt+1
    elif s[l] != s[r]: return 0, cnt+1
    else: return recursion(s, l+1, r-1, cnt+1)

def isPalindrome(s):
    return recursion(s, 0, len(s)-1)

T = int(input())
list = []
for i in range(T):
    list.append(isPalindrome(str(input())))

for i,j in list:
    print(i, j)

    

